subsys/bluetooth/audio/shell/bap.c

```diff
@@ -2381,69 +2386,109 @@ static uint16_t interval_to_sync_timeout(uint16_t interval)
+			return true;
+		}
+
+		if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+			return true;
+		}
+
+		sr_info->broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
 		return true;
-	}
+	case BT_DATA_BROADCAST_NAME:
+		if (!IN_RANGE(data->data_len, BT_AUDIO_BROADCAST_NAME_LEN_MIN,
+			      BT_AUDIO_BROADCAST_NAME_LEN_MAX)) {
+			return false;
+		}
 
-	if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+		memcpy(sr_info->broadcast_name, data->data, data->data_len);
+		sr_info->broadcast_name[data->data_len] = '\0';
+		utf8_trunc(sr_info->broadcast_name);
+		return true;
+	default:
 		return true;
 	}
+}
 
-	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
-		return true;
+static void broadcast_scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
+{
+	struct bt_broadcast_info sr_info = {0};
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	bool identified_broadcast = false;
+
+	sr_info.broadcast_id = BT_BAP_INVALID_BROADCAST_ID;
+
+	if ((auto_scan.broadcast_info.broadcast_id == BT_BAP_INVALID_BROADCAST_ID) &&
+	    (strlen(auto_scan.broadcast_info.broadcast_name) == 0U)) {
+		/* no op */
+		return;
 	}
 
-	if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO)) {
-		return true;
+	if (!passes_scan_filter(info, ad)) {
+		return;
 	}
 
-	broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+	bt_data_parse(ad, scan_check_and_sync_broadcast, (void *)&sr_info);
 
-	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	/* Verify that it is a BAP broadcaster*/
+	if (sr_info.broadcast_id == BT_BAP_INVALID_BROADCAST_ID) {
+		return;
+	}
 
-	shell_print(ctx_shell,
-		    "Found broadcaster with ID 0x%06X and addr %s and sid 0x%02X (looking for "
-		    "0x%06X)",
-		    broadcast_id, le_addr, info->sid, auto_scan.broadcast_id);
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
 
-	if (auto_scan.broadcast_id == broadcast_id && auto_scan.broadcast_sink != NULL &&
-	    auto_scan.broadcast_sink->pa_sync == NULL) {
-		struct bt_le_per_adv_sync_param create_params = {0};
-		int err;
+	if (sr_info.broadcast_id == auto_scan.broadcast_info.broadcast_id) {
+		identified_broadcast = true;
+	} else if ((strlen(auto_scan.broadcast_info.broadcast_name) != 0U) &&
+			is_substring(auto_scan.broadcast_info.broadcast_name,
+				sr_info.broadcast_name)) {
+		auto_scan.broadcast_info.broadcast_id = sr_info.broadcast_id;
+		identified_broadcast = true;
 
-		err = bt_le_scan_stop();
-		if (err != 0) {
-			shell_error(ctx_shell, "Could not stop scan: %d", err);
-		}
+		shell_print(ctx_shell, "Found matched broadcast name '%s' with address %s",
+				sr_info.broadcast_name, addr_str);
+	}
 
-		bt_addr_le_copy(&create_params.addr, info->addr);
-		create_params.options = BT_LE_PER_ADV_SYNC_OPT_FILTER_DUPLICATE;
-		create_params.sid = info->sid;
-		create_params.skip = PA_SYNC_SKIP;
-		create_params.timeout = interval_to_sync_timeout(info->interval);
+	if (identified_broadcast) {
```

Comment:
```suggestion
	if (identified_broadcast && auto_scan.broadcast_sink != NULL && auto_scan.broadcast_sink->pa_sync == NULL) {
```

We might as well couple these checks, right?